mlx_lm/generate.py

```diff
@@ -352,15 +361,22 @@ def generate_step(
 
     sampler = sampler or (lambda x: mx.argmax(x, axis=-1))
 
+    def _model_call(y):
+        if y.ndim == 3:
+            return model(None, cache=prompt_cache, input_embeddings=y)
+        else:
+            return model(y, cache=prompt_cache)
```

Comment:
I think the gain in the rest of the code simplicity outweighs the cost of using `ndim` here. I agree it's not ideal though as it's a little brittle... thanks for making that change!